src/Features/Core/Portable/DocumentationComments/CodeFixes/AbstractAddDocCommentNodesCodeFixProvider.cs

```diff
@@ -72,19 +72,30 @@ internal abstract class AbstractAddDocCommentNodesCodeFixProvider
                 // If the index is `0`, try to add the node after the `summary` node,
                 // only if any existing nodes are at the top level--this route will not
                 // be taken if the existing node is nested in another node
-                var summaryNode = GetParameterNodes(newDocComment, "summary").FirstOrDefault();
-                if (index != 0 || !paramNodes.Any() && summaryNode != null)
+                var summaryNode = GetElementNodes(newDocComment, "summary").FirstOrDefault();
+                if (index != 0 || (!paramNodes.Any() && summaryNode != null))
                 {
                     // First, try to get the node before the param node so we know where to insert the new node
-                    SyntaxNode nodeBeforeNewParamNode =  GetLastParamNodeCorrespondingToParameterInList(paramNodes, paramsBeforeCurrentParam) ??
-                                    summaryNode;
+                    TXmlElementSyntax nodeBeforeNewParamNode = null;
+                    if (index > 0)
+                    {
+                        nodeBeforeNewParamNode = GetParamNodeForParamName(paramNodes, parameterNames[index - 1]);
+                    }
+
+                    // This will be hit in the index is `0`, in which case the previous node is the summary node
+                    if (nodeBeforeNewParamNode == null)
+                    {
+                        nodeBeforeNewParamNode = summaryNode;
+                    }
+
+                    newDocComment = newDocComment.InsertNodesAfter(nodeBeforeNewParamNode,
+                        new[] { GetNewNode(parameterName, isFirstNodeInComment: false) });
 
-                    newDocComment = newDocComment.InsertNodesAfter(nodeBeforeNewParamNode, new[] { GetNewNode(parameterName, false) });
                     continue;
                 }
                 
                 // At this point, the node has to go at the beginning of the comment
-                var nodeAfterNewParamNode = GetLastParamNodeCorrespondingToParameterInList(paramNodes.AsEnumerable().Reverse(), paramsAfterCurrentParam) ??
+                var nodeAfterNewParamNode = GetFirstParamNodeCorrespondingToParameterInList(paramNodes, paramsAfterCurrentParam) ??
```

Comment:
I would just search for the first param node period.  That's the most likely place the user would want to attach this new param-node to.